src/ucs/vfs/base/vfs_obj.c

```diff
@@ -661,14 +723,15 @@ ucs_status_t ucs_vfs_path_read_file(const char *path, ucs_string_buffer_t *strb)
     ucs_spin_lock(&ucs_vfs_obj_context.lock);
 
     node = ucs_vfs_node_find_by_path(path);
-    if (!ucs_vfs_check_node(node, UCS_VFS_NODE_TYPE_RO_FILE)) {
+    if (!ucs_vfs_check_node(node, UCS_VFS_NODE_TYPE_RO_FILE) &&
+        !ucs_vfs_check_node(node, UCS_VFS_NODE_TYPE_RW_FILE)) {
```

Comment:
seems you need to check several times whether the node is `UCS_VFS_NODE_TYPE_RO_FILE` or `UCS_VFS_NODE_TYPE_RW_FILE` (the previous check with in `ucs_vfs_read_file`). Maybe create an inline func for that?